packages/cms/src/lokalize/import.ts

```
@@ -0,0 +1,136 @@
+       * otherwise moved.
+       */
+      if (typeof value !== 'string') {
+        console.warn(`Ignoring value type ${typeof value} for path ${path}`);
+        issueCounter++;
+        return;
+      }
+
+      return {
+        _type: 'lokalizeText',
+        _key: path, // _key is required for arrays in Sanity
+        path,
+        text: {
+          _type: 'localeText',
+          nl: value,
+          /**
+           * Fall back to NL text because the field is required
+           */
+          en: get(en, `${key}.${path}`, value),
+        },
+      };
+    })
+    /**
+     * Filter out the entries that got skipped
+     */
+    .filter(isDefined);
+
+  if (isEmpty(texts)) {
+    /**
+     * If no text entries survived for this key, we do not store anything
+     */
+    continue;
+  }
+
+  const document = {
+    _type: 'lokalizeSubject',
+    /**
+     * We use the key in the document id so that we can overwrite documents with
+     * updates from Lokalize. This is only required as long as we use Lokalize
+     * as our source for texts. Once we kill it, we can store these document
+     * with an opaque id which allows us to freely change the key name, to
+     * refactor the names we use in our code.
+     *
+     * The jsonKey__ prefix makes sure we don't clash with other documents in
+     * sanity that have a non-opaque document id.
+     */
+    _id: `jsonKey__${key}`,
+    key,
+    texts,
+  };
+
+  queue.add(() =>
+    client
+      /**
+       * Strings from JSON will be overwriting whatever is in Sanity, because
+       * until we kill Lokalize those files are the source of truth. However,
+       * newly created text entries in Sanity will remain untouched, so we could
+       * already start to extend the Lokalize string set using Sanity if we want
+       * to.
+       */
+      .createOrReplace(document)
+      .catch((err) => {
+        console.error(
+          `Failed to create document for key ${key}: ${err.message}`
+        );
```

Do we want the import/export scripts to fail silently or should it actually throw an error to block build scripts?

---

or would console.error result in the same behavior as a throw?

---

Import is something we do manually and only a few times until we hit the killswitch. So I think error messages in the console are fine.